### PR TITLE
Added get wallet by username API

### LINKED ISSUE #281: GET /wallet/:username
Fetch wallet for a given username.

### PR DESCRIPTION
**Pull request by ankushdharkar**
Closes #281

This API is required for superusers to fetch wallets of any user, so that they can show it on a dashboard: https://github.com/Real-Dev-Squad/website-dashboard/pull/16

Tests are missing, but tracking it here: https://github.com/Real-Dev-Squad/website-backend/issues/283

### CODE REVIEW

File: controllers/walletsController.js [hunk 1]
```diff
@@ -22,6 +23,28 @@ const getUserWallet = async (req, res) => {
...
 }
 
+/**
+ * Get the wallet details of user, if a username is provided
+ * @param {Object} req - Express request object
+ * @param {Object} res - Express response object
+ */
+const getWallet = async (req, res) => {
+  try {
+    const { params: { username } = {} } = req
```
Comment (ankurnarkhede) [inline controllers/walletsController.js:33]: I presume `user1` will be allowed to get the wallet data of only `user1`.
Here, `user1` can pass the username of `user2` and get their wallet details!

Thus, we should have validation if the username passed with path params corresponds to the cookie owner.
We can get the authenticated user's data using the object at `req.userData`.

Thus, the `getUserId` utility is not required here.

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: routes/wallets.js [hunk 2]
```diff
@@ -32,4 +33,33 @@ const authenticate = require('../middlewares/authenticate')
...
+ *     tags:
+ *       - wallet
+ *     responses:
+ *       200:
+ *         description: Return wallet
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/wallet'
+ *       401:
```
[the remark this answers is not in the document]
Comment (ankurnarkhede) [reply routes/wallets.js:50]: 403 will be thrown by authorization middleware.

Can you also generate the API schema by running the command given at: https://github.com/Real-Dev-Squad/website-backend/blob/develop/README.md#api-documentation

[the remark this answers is not in the document]
Comment (ankushdharkar) [reply routes/wallets.js:50]: Awesome catch! Thanks. Adding 403.

File: routes/wallets.js [hunk 3]
```diff
@@ -32,4 +33,33 @@ const authenticate = require('../middlewares/authenticate')
...
+ *             schema:
+ *               $ref: '#/components/schemas/errors/unAuthorized'
+ *       500:
+ *         description: badImplementation
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/errors/badImplementation'
+ */
+router.get('/:username', authenticate, authorization, walletController.getWallet)
```
[the remark this answers is not in the document]
Comment (ankushdharkar) [reply routes/wallets.js:63]: Makes sense. Updating it.

File: controllers/walletsController.js (hunk 1 above)
Comment (ankushdharkar) [reply controllers/walletsController.js:40]: We are authorizing this route for super users. So, it should not be a problem.

**Review by ankurnarkhede — APPROVED** [review]
LGTM ✅

**Pull request merged**